### PR TITLE
feat(swap): anti-rekt streaming fallback for THORChain quotes

### LINKED ISSUE #4205: [ADD] Anti-rekt: auto-fallback to THORChain streaming swap when rapid slippage exceeds 3%
## Problem

THORChain rapid swaps (`interval=0`) on shallow pools can inflict catastrophic slippage that surfaces as `fees.total` in the quote. Observed in-app:

- **1 BTC → TRX**: input $75,766, output $24,079 → **$43,440 (57%) "Swap Fee"** from the new shallow TRON pool
- **1 BTC → XRP**: input $75,766, output $55,720 → **$12,504 (16%) "Swap Fee"**

Both numbers are legitimate `fees.total` values from the THORChain quote endpoint — the AMM's liquidity fee scales with trade-size vs pool-depth. Users have no in-wallet protection. A comparable `[ADD] Max slippage for Lifi/1inch swaps` was shipped in 2024 (#1601) for EVM providers; no equivalent exists for THORChain.

## Prior context

The rapid-vs-streaming decision has deliberate history:
- iOS #4088 (closed) — THORChain `interval` switched from 1 → 0
- Android #3702 (closed) — same
- Windows #3613 (open) — same switch still pending

The switch to rapid was **correct**: streaming swaps take multiple blocks (5-30+ minutes) and the "confirm → 20 minutes of silence" UX confused users. Rapid is right for ~99% of trades. **This issue does not reverse that decision — it adds an anti-rekt escape hatch** that keeps rapid as default and only reaches for streaming when the user would otherwise eat >3% slippage on a shallow pool.

## Proposed solution

In the THORChain quote fetch path (single-file scope):

1. Fetch rapid quote as today (`interval=0`). — **Normal path. No behavior change for ~99% of trades.**
2. Compute `rapidSlippageBps = (fees.total_in_fiat / input_fiat) × 10_000`.
3. If `rapidSlippageBps ≤ 300` (3%): return rapid. Done.
4. Else fetch a streaming quote with `interval=1, streaming_quantity=rapid.max_streaming_quantity` (the quote response's own optimal suggestion).
5. If streaming fetch fails: log and return rapid (silent fallback — user sees current behavior, no new error surface).
6. If `streaming.expected_amount_out > rapid.expected_amount_out`: return streaming. Else: rapid.

The `memo` field in the quote response already embeds `/INTERVAL/QUANTITY` when streaming params are requested. **Signing code remains unchanged** — verified that `quote.memo` passes through verbatim in `SwapCryptoLogic.swift`.

## Scope

**In scope**
- Threshold-based double-fetch + quote selection in the THORChain quote repository.
- Single configurable constant `streamingSlippageThresholdBps = 300`.
- Unit tests for the four branches (below threshold, above + streaming better, above + streaming worse, streaming fetch error).

**Out of scope (explicit)**
- **No UI changes.** No rapid/streaming toggle, no ETA label, no streaming badge. Users always see the best available quote. Execution mode is observable via tx history / on-chain.
- No time cap on streaming duration — trust THORChain's `max_streaming_quantity`.
- Maya provider (same quote shape; separate ticket if we want parity).
- Telemetry (follow-up).

## Acceptance

- [ ] Rapid `fees.total ≤ 3%` → streaming quote NOT fetched (normal path unchanged)
- [ ] Rapid `fees.total > 3%` → streaming quote fetched
- [ ] Streaming fetch exception → rapid returned, no exception to UI
- [ ] `streaming.expected_amount_out > rapid` → streaming memo (`/INTERVAL/QUANTITY`) surfaces to signing
- [ ] `streaming.expected_amount_out ≤ rapid` → rapid returned
- [ ] Unit tests for all four branches
- [ ] Manual QA: **1 BTC → TRX** auto-switches to streaming with meaningful output improvement; **0.01 BTC → ETH** stays rapid unchanged

## Risk / Rollback

- One constant. Raise to `.max` to fully disable without reverting code.
- At most one extra THORChain API call per quote refresh, only when slippage > 3%.
- Streaming fetch failure → silent fallback → rapid. No new error surface.
- Rollback: single `git revert`. No schema, no persisted state.

## Platform parity

Sibling tickets opened on `vultisig/vultisig-android` and `vultisig/vultisig-windows` with the same spec. Signing-side memo handling is already consistent across all three platforms, so only the quote-fetch layer changes per repo.

### PR DESCRIPTION
**Pull request by gastonm5**
## Summary

THORChain rapid swaps (`streaming_interval=0`) on shallow pools can inflict catastrophic slippage that surfaces as `fees.total` in the quote — e.g. **1 BTC → TRX losing 57%** to the new shallow TRON pool. This adds an automatic fallback: when rapid `fees.total` exceeds 3% of the gross output, we re-fetch the quote as a streaming swap and pick whichever has the higher `expected_amount_out`.

Rapid remains the default for ~99% of trades; streaming is reached for only when the user would otherwise eat >3% slippage. No UI changes — the best quote is chosen silently.

## Changes

- `ThorchainSwapQuote` gains `max_streaming_quantity` so we can forward THORChain's own optimal streaming suggestion.
- `Endpoint.fetchSwapQuoteThorchain` threads an optional `streaming_quantity` query param.
- `ThorchainSwapProvider` protocol (and the 4 implementations: Thorchain, Chainnet, Stagenet, Maya) accept `streamingQuantity`.
- `SwapService.fetchCrossChainQuote` adds the fallback path, gated by a single constant `streamingSlippageThresholdBps = 300`.
- Silent fallback to rapid if the streaming fetch throws.
- Maya is excluded from the fallback (out of scope).
- Unit tests for the four acceptance branches (below threshold, above + streaming better, above + streaming worse, streaming-fetch error) plus edge cases.

## Acceptance

- [x] Rapid `fees.total ≤ 3%` → streaming quote NOT fetched
- [x] Rapid `fees.total > 3%` → streaming quote fetched with `streaming_interval=1` and `streaming_quantity=max_streaming_quantity`
- [x] Streaming fetch exception → rapid returned, no exception to UI
- [x] `streaming.expected_amount_out > rapid` → streaming returned (memo passes through verbatim to signing)
- [x] `streaming.expected_amount_out ≤ rapid` → rapid returned
- [x] Unit tests cover all four branches

## Test plan

- [x] `swiftlint lint --config VultisigApp/.swiftlint.yml` — 0 violations
- [x] `xcodebuild build` — succeeds (iPhone 17 simulator)
- [x] `xcodebuild test -only-testing:VultisigAppTests/ThorchainAntiRektTests` — 12/12 pass
- [ ] Manual QA: 1 BTC → TRX auto-switches to streaming; 0.01 BTC → ETH stays rapid

## Risk / Rollback

- Single constant controls behavior; raise to `.max` to effectively disable.
- At most one extra quote call per refresh, and only when slippage > 3%.
- Streaming fetch failure → silent fallback → rapid. No new error surface.
- Rollback: `git revert`. No schema, no persisted state.

## Related

- Closes #4205

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

## Summary by CodeRabbit

* **New Features**
  * Added anti-REKT mechanism for THORChain swaps that automatically fetches streaming quotes when price impact exceeds threshold, improving slippage protection and quote selection.
  * Enhanced swap quote support with additional fee and streaming information fields.

* **Tests**
  * Added comprehensive test coverage for anti-REKT swap protection, including slippage calculations and quote comparisons.

* **Chores**
  * Updated dependency.

### CODE REVIEW

File: VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapService.swift [hunk 1]
```diff
@@ -290,3 +309,110 @@ private extension SwapService {
...
+
+        let slippageBps = Self.rapidSlippageBps(fromQuote: rapid) ?? 0
+        logger.info("[anti-rekt] rapid slippage=\(slippageBps, privacy: .public) bps, threshold=\(Self.streamingSlippageThresholdBps, privacy: .public) bps, fromAsset=\(fromAsset, privacy: .public), toAsset=\(toAsset, privacy: .public)")
+
+        guard slippageBps > Self.streamingSlippageThresholdBps else {
+            logger.info("[anti-rekt] slippage ≤ threshold → using RAPID (memo=\(rapid.memo, privacy: .public))")
+            return rapid
+        }
+
+        let streamingQuantity = rapid.maxStreamingQuantity ?? 0
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapService.swift:391]: I think using streaming Quantity is not correct

When trying to swap 0.5 BTC -> TRON.TRX , the slippage is 4021 bps , but it didn't fallback to streaming swaps 

We should ignore streaming Quantity check, and we don't use stream quantity before , it usually it is 0

**Review by johnnyluo — CHANGES_REQUESTED** [review]
see comments

File: VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapService.swift (hunk 1 above)
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapService.swift:391]: It should be good now, it uses RAPID up to ~0.03 BTC, then streaming

**Review by johnnyluo — APPROVED** [review]
LGTM
